src/sdl_utils.cpp

```diff
@@ -0,0 +1,57 @@
+#include "sdl_gl.h"
+#include <iostream>
+
+void sdl_application_abort(const char * msg)
+{
+    std::cout << SDL_GetError() << std::endl;
+    SDL_Quit();

```

Comment:
take a look at main.cpp.
I don't normally use exceptions, but the main application has a final catch_all for exceptions which shuts the engine down properly.

I recommend:
    throw(1);

Like in the initialization error checks, which the main app will catch and shutdown the engine including "SDL_Quit();"